src/components/contextual/pages/pool/StakingIncentivesCard/StakingIncentivesCard.vue

```diff
@@ -0,0 +1,232 @@
+<script setup lang="ts">
+import useNumbers, { FNumFormats } from '@/composables/useNumbers';
+import { computed, onBeforeMount, ref } from 'vue';
+
+import AnimatePresence from '@/components/animate/AnimatePresence.vue';
+import useStaking from '@/composables/staking/useStaking';
+import { FullPool } from '@/services/balancer/subgraph/types';
+import { bnum } from '@/lib/utils';
+import useTokens from '@/composables/useTokens';
+import { getAddress } from 'ethers/lib/utils';
+import StakePreview from '../../../stake/StakePreview.vue';
+
+type Props = {
+  pool: FullPool;
+};
+const props = defineProps<Props>();
+/**
+ * STATE
+ */const isLoading = ref(false);
+const hasIncentive = true;
+const isStakePreviewVisible = ref(false);
+const stakeAction = ref('');
+const stakedShares = ref('0');
+
+/**
+ * COMPOSABLES
+ */
+const { fNum2 } = useNumbers();
+const { balanceFor } = useTokens();
+const { isFetchingStakingData, getStakedShares } = useStaking(
+  props.pool.address
+);
+
+/**
+ * COMPUTED
+ */
+// TODO INTEGRATE STAKING APR
+const apr = computed(() => {
+  return 0.1245;
+});
+
+const fiatValueOfStakedShares = computed(() => {
+  return bnum(props.pool.totalLiquidity)
+    .div(props.pool.totalShares)
+    .times(stakedShares.value.toString())
+    .toString();
+});
+
+const fiatValueOfUnstakedShares = computed(() => {
+  return bnum(props.pool.totalLiquidity)
+    .div(props.pool.totalShares)
+    .times(balanceFor(getAddress(props.pool.address)))
+    .toString();
+});
+
+/**
+ * METHODS
+ */
+function showStakePreview() {
+  if (fiatValueOfUnstakedShares.value === '0') return;
+  stakeAction.value = 'stake';
+  isStakePreviewVisible.value = true;
+}
+
+function showUnstakePreview() {
+  if (fiatValueOfStakedShares.value === '0') return;
+  stakeAction.value = 'unstake';
+  isStakePreviewVisible.value = true;
+}
+
+function handlePreviewClose() {
+  isStakePreviewVisible.value = false;
+}
+
+async function handleActionSuccess() {
+  stakedShares.value = await getStakedShares();
+}
+
+/**
+ * LIFECYCLE
+ */
+onBeforeMount(async () => {
+  stakedShares.value = await getStakedShares();
```

Comment:
Receiving `stakedShares` from contract pending response from Fabio regarding whether `balance` from gauge subgraph is cumulative or not.